mindocr/utils/misc.py

```diff
@@ -35,10 +35,18 @@ def fetch_optimizer_lr(opt):
         if opt.is_group_lr:
             lr = ()
             for learning_rate in opt.learning_rate:
-                cur_dynamic_lr = learning_rate(opt.global_step - 1).reshape(())
+                cur_dynamic_lr = (
+                    learning_rate(opt.global_step - 1).reshape(())
+                    if ms.version.__version__ < "2.2"
+                    else learning_rate(opt.global_step).reshape(())
+                )
                 lr += (cur_dynamic_lr,)
         else:
-            lr = opt.learning_rate(opt.global_step - 1).reshape(())
+            lr = (
+                opt.learning_rate(opt.global_step - 1).reshape(())
+                if ms.version.__version__ < "2.2"
+                else opt.learning_rate(opt.global_step).reshape(())
```

Comment:
should keep `opt.global_step - 1` for version >= 2.2, since the learning rate list index starts from 0

Reply:
The PR has been revised according to this review comments. The attribute `self.global_step` does not increase automatically for MindSpore >= r2.2.0, So the codes above was added to avoid IndexError. But it is unnecessary  because `self.global_step` += 1 has been added to the optimizer classes.